File: 백준/Silver/11055. 가장 큰 증가하는 부분 수열/가장 큰 증가하는 부분 수열.py
import sys

def solution():
  N = int(sys.stdin.readline())
  numbers = list(map(int, sys.stdin.readline().split()))
  dp = [1 for _ in range(N)]
  dp[0] = numbers[0]

  for i in range(1, N):
    for j in range(i):
      if numbers[i] > numbers[j]:
        dp[i] = max(dp[i], dp[j] + numbers[i])
      else:
        dp[i] = max(dp[i], numbers[i])
  
  print(max(dp))
    
solution()

File: 백준/Silver/11055. 가장 큰 증가하는 부분 수열/가장 큰 증가하는 부분 수열.js
let fs = require('fs');
let inputs = fs.readFileSync('/dev/stdin').toString().split('\n');

const N = Number(inputs[0]);
const A = inputs[1].split(' ').map(Number);
const dp = [A[0]];

for (let i = 1; i < A.length; i++) {
    let max = 0;
    for (let j = i - 1; j >= 0; j--) {
        if (A[j] < A[i] && max < dp[j]) {
            max = dp[j]
        }
    }
    max !== 0 ? dp.push(max + A[i]) : dp.push(A[i]);
}

console.log(Math.max(...dp));